### ISSUE TITLE
Memory control

### ISSUE THREAD

**Issue by jakewendt** [RitchieLabIGH/iMOKA#12]
Hello again Claudio.

I am running this pipeline on our cluster and I continue to increase k to see how far I can go. Initially, I set IMOKA_MAX_MEM_GB to the full amount of memory requested for the job minus about 10GB for singularity headroom. However, the higher I go with k, the lower I have to set `IMOKA_MAX_MEM_GB` because the job would crash with OUT_OF_MEMORY. For example, if I submit a job that requests 450GB of memory, I have to set IMOKA_MAX_MEM_GB=256 for k=81. If I increase k to 91, I have to lower IMOKA_MAX_MEM_GB to 192.

I thought that I'd have a look at memory usage off of the cluster. When I run ...
```
export OMP_NUM_THREADS=20
export IMOKA_MAX_MEM_GB=100
singularity exec  iMOKA_extended-1.1.2.img iMOKA_core reduce --input matrix.json --output testreduce.matrix > testreduce.out 2> testreduce.err &
```
... for my dataset with k=31, it actually peaks at about 90GB.

For k=41 it actually peaks at about 115GB.

For k=61 it actually peaks at about 155GB.

Is there a better way to more accurately and more stably set the memory limit?

Thanks
Jake

**Comment by RitchieLabIGH** [comment]
Hello Jake,
the estimation of the memory usage was based on k=31, that's why with different sizes it doesn't fit.
I just modified the method that takes care of it. Since I also upgraded some libraries, I'm going to check that everything works smoothly before committing. I'll leave another comment here as soon as it's done.
Thank you for your feedback: as usual, it was very helpful.
Cheers,
Claudio
